### PR TITLE
Remove jax.ops.index... functions.

### PR DESCRIPTION
**Pull request by hawkinsp**
These functions have been deprecated and have issued a DeprecationWarning since jax 0.2.22 in October 2021.

### CODE REVIEW

**Comment by hawkinsp** [comment]
I suspect it will be a week or so before I can land this; I still need to make a number of cleanups.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks good! We'll also need to remove references from https://github.com/google/jax/blob/main/docs/jax.ops.rst and perhaps other places in the docs

**Comment by hawkinsp** [comment]
> Looks good! We'll also need to remove references from https://github.com/google/jax/blob/main/docs/jax.ops.rst and perhaps other places in the docs

Done.

File: docs/jax.ops.rst [hunk 1]
```diff
@@ -1,75 +1,15 @@
...
-=================
+===============
 
 .. currentmodule:: jax.ops
 
 .. automodule:: jax.ops
 
 .. _syntactic-sugar-for-ops:
 
-Indexed update operators
```
Comment (jakevdp) [inline docs/jax.ops.rst:11]: Keep lines 11-52?

Comment (hawkinsp) [reply docs/jax.ops.rst:11]: They have no reason to be in this particular module once the deprecation is complete. Is the reason to point people to the new names?

Comment (hawkinsp) [reply docs/jax.ops.rst:11]: I added a paragraph that points to jnp.ndarray.at.

**Comment by gnecula** [comment]
It is rare to see a case when we simplify the API! Thanks!

**Pull request merged**